python/unitytrainers/meta_curriculum.py

```diff
@@ -0,0 +1,85 @@
+"""Contains the MetaCurriculum class."""
+
+import os
+from unitytrainers import Curriculum
+
+class MetaCurriculum(object):
+    """A MetaCurriculum holds curriculums. Each curriculum is associated to a particular
+    brain in the environment.
+    """
+
+    def __init__(self, curriculum_folder, default_reset_parameters):
+        """Initializes a MetaCurriculum object.
+
+        Args:
+            curriculum_folder (str): The relative or absolute path of the
+                folder which holds the curriculums for this environment.
+                The folder should contain JSON files whose names are the
+                brains that the curriculums belong to.
+            default_reset_parameters (dict): The default reset parameters
+                of the environment.
+        """
+        if curriculum_folder is None:
+            self._brains_to_curriculums = None
+        else:
+            self._brains_to_curriculums = {}
+            for curriculum_filename in os.listdir(curriculum_folder):
+                brain_name = curriculum_filename.split('.')[0]
+                curriculum_filepath = \
+                    os.path.join(curriculum_folder, curriculum_filename)
+                self._brains_to_curriculums[brain_name] = \
+                    Curriculum(curriculum_filepath, default_reset_parameters)
+
+    @property
+    def brains_to_curriculums(self):
+        """A dict from brain_name to the brain's curriculum."""
+        return self._brains_to_curriculums
+
+    @property
+    def lesson_nums(self):
+        """A dict from brain name to the brain's curriculum's lesson number."""
+        lesson_nums = {}
+        for brain_name, curriculum in self.brains_to_curriculums.items():
+            lesson_nums[brain_name] = curriculum.lesson_num
+
+        return lesson_nums
+
+    @lesson_nums.setter
+    def lesson_nums(self, lesson_nums):
+        for brain_name, lesson in lesson_nums.items():
+            self.brains_to_curriculums[brain_name].lesson_num = lesson
+
+    def increment_lessons(self, progresses):
+        """Increments all the lessons of all the curriculums in this MetaCurriculum.
+
+        Args:
+            progresses (dict): A dict of brain name to progress.
+        """
+        for brain_name, progress in progresses.items():
+            self.brains_to_curriculums[brain_name].increment_lesson(progress)
+
+
+    def set_all_curriculums_to_lesson_num(self, lesson_num):
```

Comment:
Do we want to be able to set all lessons to the same number or do we want to be able to set the lessons individually when using learn.py ?

Reply:
Since `lesson_nums` is a property it's easy to set all the lesson numbers individually. This function just makes it convenient for initializing all the lessons to a specific one.